homeassistant/components/atag/climate.py

```diff
@@ -0,0 +1,132 @@
+"""Initialization of ATAG One climate platform."""
+from typing import List, Optional
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    DEFAULT_MAX_TEMP,
+    DEFAULT_MIN_TEMP,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    PRESET_BOOST,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from . import CLIMATE, DOMAIN, ENTITY_TYPES, AtagEntity
+
+PRESET_SCHEDULE = "Auto"
+PRESET_MANUAL = "Manual"
+PRESET_EXTEND = "Extend"
+SUPPORT_PRESET = [
+    PRESET_MANUAL,
+    PRESET_SCHEDULE,
+    PRESET_EXTEND,
+    PRESET_BOOST,
+]
+SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+
+
+async def async_setup_entry(hass, entry, async_add_devices):
+    """Load a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_devices([AtagThermostat(coordinator, ENTITY_TYPES[CLIMATE])])
+
+
+class AtagThermostat(AtagEntity, ClimateDevice, RestoreEntity):
+    """Atag climate device."""
+
+    def __init__(self, coordinator, atagtype):
+        """Initialize with fake on/off state."""
+        self._on = None
+        super().__init__(coordinator, atagtype)
+
+    async def async_added_to_hass(self):
+        """Register callbacks & state restore for fake "Off" mode."""
+        await super().async_added_to_hass()
+        last_state = await self.async_get_last_state()
+        if last_state:
+            self._on = last_state.state != HVAC_MODE_OFF
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def hvac_mode(self) -> Optional[str]:
+        """Return hvac operation ie. heat, cool mode."""
+        if not self._on:
+            return HVAC_MODE_OFF
+        return self.coordinator.atag.hvac_mode
```

Comment:
Only the home assistant climate constants for HVAC mode are allowed to be return here.
https://github.com/home-assistant/core/blob/d144228272deff59e86b34418d67cc964a956771/homeassistant/components/climate/const.py#L25-L33

Reply:
the library actually does return values in this list (AUTO and HEAT), or should I use a mapping list?

Reply:
added a line to ensure no incorrect hvac modes are returned 👍